server/core/src/main/java/org/infinispan/server/core/transport/NettyChannelInitializer.java

```diff
@@ -28,19 +28,24 @@
    protected final NettyTransport transport;
    protected final ChannelOutboundHandler encoder;
    protected final ChannelInboundHandler decoder;
+   private final RuleBasedIpFilter iprulesHandler;
+   private final StatsChannelHandler statsHandler;
 
    public NettyChannelInitializer(ProtocolServer<A> server, NettyTransport transport, ChannelOutboundHandler encoder, ChannelInboundHandler decoder) {
       this.server = server;
       this.transport = transport;
       this.encoder = encoder;
       this.decoder = decoder;
+      this.statsHandler = transport != null ? new StatsChannelHandler(transport) : null;
+      this.iprulesHandler = new RuleBasedIpFilter(server.getConfiguration().ipfilter());
    }
 
    @Override
    public void initializeChannel(Channel ch) throws Exception {
       ChannelPipeline pipeline = ch.pipeline();
-      if(transport != null) {
-         pipeline.addLast("stats", new StatsChannelHandler(transport));
+      pipeline.addLast("iprules", iprulesHandler);
```

Comment:
shouldn't the ip filter be the first one?

Reply:
Tues